woocommerce/woocommerce-gutenberg-products-block
# Try: Set up plugin to publish on npm

Fixes #433. 

This should build a package with the following files, which can be installed on a site as-is and run (this helps build the zip we attach to each release).

```
package.json
license.txt
README.md
readme.txt
woocommerce-gutenberg-products-block.php
assets/*
build/*
includes/*
languages/*
```

The biggest change here is to create the `assets/php` directory which contains the PHP files that will be imported into woocommerce core. The new `WGPB_Block_Library` class handles registering the blocks and all associated with that (scripts, styles, the WooCommerce block category, and the global settings).

The featured product handler class is also in this folder, renamed `WGPB_Block_Featured_Product`. Both of these files are processed in the import into woocommerce core such that all "WGPB" references are converted to "WC". [See the grunt copy step in that branch.](https://github.com/woocommerce/woocommerce/blob/5eca4eb302f96303e142e6dc83ce03c5fa814f67/Gruntfile.js#L372-L386) This prevents "class X is already registered" errors, and lets us override the core files so that we can keep iterating in this plugin.

### To test

1. You can run `npm pack` to generate a `.tgz` file of what would be pushed to npm (or `npm pack --dry-run` to see the list without creating the file)

To test that we're correctly overriding the core files:

1. Deactivate this plugin
2. Check out [the blocks branch on woocommerce](https://github.com/woocommerce/woocommerce/pull/22858)
3. Check that the files are loading from woocommerce core by view-source on the editor, look for `product-new.js`
4. The file should be there, and the source should be `woocommerce/assets/js/blocks/product-new.js`
5. Activate the feature plugin
6. view-source again, the source should now be `/woocommerce-block-library/build/product-new.js`
7. Run `npm start` in the feature plugin
8. Make some visible change to a block
9. Check in the editor that that change renders

To check that we're not breaking back-compat:

1. Using WooCommerce 3.5 (`git checkout 359d1130` in woocommerce will get you 3.5.5), run this branch
2. Check that all blocks load as expected.
